akka-stream/src/main/scala/akka/stream/impl/FlexiMergeImpl.scala

```diff
@@ -44,7 +44,10 @@ private[akka] class FlexiMergeImpl(_settings: MaterializerSettings,
 
   override protected val inputBunch = new FanIn.InputBunch(inputPorts, settings.maxInputBufferSize, this) {
     override def onError(input: Int, e: Throwable): Unit = {
-      changeBehavior(completion.onError(ctx, inputMapping(input), e))
+      changeBehavior(try completion.onError(ctx, inputMapping(input), e)
+      catch {
+        case NonFatal(e) ⇒ fail(e); mergeLogic.SameState
```

Comment:
no, inputBunch and primaryOutputs have protection flag for that, so it is idempotent